### PR TITLE
Populate random values to a specific tensor

### PR DESCRIPTION
**Pull request by yiqian1**
Implement [issue#147](https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/147)

### CODE REVIEW

File: mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_random.mlir [hunk 1]
```diff
@@ -72,3 +72,14 @@
 
 // GROUP_NHWC: Unranked Memref base@ = 0x{{.*}} rank = 1 offset = 0 sizes = [1] strides = [1] data =
 // GROUP_NHWC: [1]
+
+mlir-miopen-driver -pv -p -rand 1 -rand_side filter -c| mlir-rocm-runner  --shared-libs=%rocm_wrapper_library_dir/librocm-runtime-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext --entry-point-result=void | FileCheck %s --check-prefix=FWD_FILTER
```
Comment (whchung) [inline mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_random.mlir]: this is incorrect. missing RUN macro

**Review by whchung — CHANGES_REQUESTED** [review]

File: mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_random.mlir [hunk 2]
```diff
@@ -72,3 +72,14 @@
...
+
+// FWD_FILTER: Unranked Memref base@ = 0x{{.*}} rank = 1 offset = 0 sizes = [1] strides = [1] data =
+// FWD_FILTER: [1]
+
+mlir-miopen-driver -pv -p -rand 1 -rand_side input -c| mlir-rocm-runner  --shared-libs=%rocm_wrapper_library_dir/librocm-runtime-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext --entry-point-result=void | FileCheck %s --check-prefix=FWD_INPUT
```
Comment (whchung) [inline mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_random.mlir]: same here. it's not going to run.

File: mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp [hunk 3]
```diff
@@ -246,6 +246,12 @@ static cl::opt<std::string> randomData(
         "use all ones. Otherwise, use time(0) as the seed."),
     cl::value_desc("seed"), cl::init("none"));
 
+static cl::opt<std::string>
+    randomSide("rand_side",
+               cl::desc("To populate random numbers to a specified tensor, "
+                        "e.g. -rand_side filter"),
```
Comment (whchung) [inline mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp]: Please enumerate allowable inputs. Also, `both` is not specified.

File: mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp [hunk 4]
```diff
@@ -1119,6 +1125,120 @@ static std::tuple<short, short, int> configRandomTestData() {
...
+    mlir::Value &outputMemsetMinValue, mlir::Value &outputMemsetMaxValue,
+    mlir::ConstantOp &seedConstantIntOp) {
+  auto int16Type = builder.getIntegerType(16);
+  auto int32Type = builder.getIntegerType(32);
+  unsigned short zero = 0, one = 1;
+  short min, max;
+  int seed = 1;
+  std::tie(min, max, seed) = configRandomTestData();
+
+  mlir::ConstantOp oneConstantIntOp;
```
Comment (whchung) [inline mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp:1141]: we could just set `minConstantIntOp` and `maxConsnatIntOp` be `1` and reduce the logic a little bit, by eliminating `oneConstantIntOp`.

File: mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp [hunk 5]
```diff
@@ -246,6 +246,12 @@ static cl::opt<std::string> randomData(
         "use all ones. Otherwise, use time(0) as the seed."),
     cl::value_desc("seed"), cl::init("none"));
 
+static cl::opt<std::string>
+    randomSide("rand_side",
+               cl::desc("To populate random numbers to a specified tensor, "
+                        "e.g. -rand_side filter"),
+               cl::value_desc("tensor"), cl::init("both"));
```
Comment (whchung) [inline mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp]: e2e test case using `both` is not provided.

**Review by whchung — CHANGES_REQUESTED** [review]
Additional tests and documentation is needed. Other than that the PR looks fine.

File: mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp (hunk 5 above)
Comment (yiqian1) [reply mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp]: "both" is a default value. All existing random tests belong to this case.

File: mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp [hunk 6]
```diff
@@ -246,6 +246,13 @@ static cl::opt<std::string> randomData(
         "use all ones. Otherwise, use time(0) as the seed."),
     cl::value_desc("seed"), cl::init("none"));
 
+static cl::opt<std::string> randomSide(
+    "rand_side",
+    cl::desc("To populate random numbers to a specified tensor: "
+             "-rand_side filter, -rand_side input, -rand_side output. "
+             "By default populate random numbers to both tensors."),
```
Comment (whchung) [inline mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp]: Allowable inputs for different `operation` is not specified.

**Review by whchung — CHANGES_REQUESTED** [review]
additional documentation on the proper semantic wrt `operation` is needed.

**Review by whchung — APPROVED** [review]
LGTM

**Pull request merged**